Add ability to delete a service from a graph

JakeDawkins opened the pull request:
It works! The `partial delete` previews the composition errors, asks for confirmation, deletes the service, and gives an output of composition errors post-delete. 

The output here looks really disgusting though. Post-RC, we should _definitely_ clean this up.

Todo: 
- [x] change the `error` pre-delete to a warning

![image](https://user-images.githubusercontent.com/9259509/99697616-e65b4b80-2a5d-11eb-9733-c87784e36194.png)

JakeDawkins committed:
add ability to delete services from graph

JakeDawkins committed:
lint

JakeDawkins committed:
change error to warning pre-delete

JakeDawkins committed:
Merge branch 'main' into jake/delete

JakeDawkins committed:
pass client to `run` by reference

EverlastingBugstopper commented on src/command/partial/delete.rs:
```diff
@@ -0,0 +1,184 @@
+
+    /// Skip the delete preview as well as the prompt confirming deletion
+    #[structopt(long)]
+    confirm: bool,
```
is this just to ignore composition errors? i think the wording of the help text is slightly confusing

JakeDawkins replied:
It's skips the "Confirm" step, where it asks the user if they'd like to proceed with deletion. That step shows a preview of composition errors that would occur if they confirmed the deletion

EverlastingBugstopper commented on crates/rover-client/src/query/partial/delete.rs:
```diff
@@ -0,0 +1,160 @@
+        Some(data) => Ok(data),
+        None => Err(RoverClientError::HandleResponse {
+            msg: "No service found".to_string(),
+        }),
```
this is a case where i would add a new variant to `RoverClientError`. so instead of `Err(RoverClientError::HandleResponse( { msg: "Stringly typed message" }`, I'd do `Err(RoverClientError::NoServiceFound)` with the message you want in the `#[error("message")]` portion

JakeDawkins replied:
That makes sense!

EverlastingBugstopper reviewed (approved):
couple small things, but overall this is looking really good

JakeDawkins committed:
Merge branch 'main' into jake/delete

JakeDawkins committed:
Update errors & docs

- added a NoService error that unifies all the disparate errors for
  "no service found. check api key/graph/variant"
- Update flag docs for partial delete to be more clear

JakeDawkins committed:
lint
